### DISCUSSION TITLE
Applying .meta directives to new files

### DISCUSSION THREAD

**Discussion by jbanana** [mbrubeck/agate#141] [Q&A] [answered]
I have files whose names begin with underscore and with no extension. I want them served as `text/gemini` so I added this to the relevant `.meta` file:
`_*: text/gemini`

This works, but if I create new files matching that pattern, they are served as `application/octet-stream`. When I re-start Agate, the new files switch to being served as `text/gemini`.

Is there a way round this besides restarting?

**Comment by Johann150** [comment, marked as answer]
The issue here is that the Globs (`*` or `**`) are "compiled" (i.e. expanded into all matching absolute paths) when the `.meta` file is read. This allows Agate to just look up the path, without having to handle pattern matching or conflicts while serving a file. This of course means better performance no matter how complicated your configuration file is. (The metadata lookup performance is *O(log(n))* where *n* is the number of entries compiled to.)

However, the `.meta` file will be reread during a request, if its last modified time changes, which also recompiles the globs. And that could be done by either editing and saving the file, or by using touch(1).

**Reply by jbanana** [reply]
Thanks. Touching the `.meta` file fixes the problem.
